srcpkgs/cnrdrvcups-sfp/template

```diff
@@ -0,0 +1,219 @@
+	vsed -e '2a export LIBS="-latk-1.0 -lgobject-2.0 -lglib-2.0 -lgdk_pixbuf-2.0 -lcups"' -i "${_driver_dir}/StatusMonitor/autogen.sh"
+
+	for _subdir in "buftool" "cngplp" "cnjbig" "rasterfilter"
+	do
+		( cd ${_common_dir}/${_subdir} && autoreconf -i )
+	done
+
+	for _subdir in "cngplp/files" "cngplp" "cpca" "ppd" "StatusMonitor"
+	do
+		( cd ${_driver_dir}/${_subdir} && autoreconf -i )
+	done
+}
+
+do_configure() {
+	local _buftool_abs_path="${wrksrc}/${build_wrksrc}/${_common_dir}/buftool"
+
+	( cd ${_common_dir}/buftool && ./autogen.sh ${configure_args} --enable-progpath=/usr/bin --disable-shared --enable-static )
+	( cd ${_common_dir}/cngplp && ./autogen.sh ${configure_args} COMMON_SUFFIX=2l )
+	( cd ${_common_dir}/cnjbig && ./autogen.sh ${configure_args} )
+	( cd ${_common_dir}/rasterfilter && LDFLAGS="-L${_buftool_abs_path} -no-pie" ./autogen.sh ${configure_args} )
+
+	( cd ${_driver_dir}/ppd && ./autogen.sh ${configure_args} )
+	( cd ${_driver_dir}/cngplp && ./autogen.sh ${configure_args} --prefix=/usr )
+	( cd ${_driver_dir}/cngplp/files && ./autogen.sh ${configure_args} )
+	( cd ${_driver_dir}/StatusMonitor && LDFLAGS="-L${_buftool_abs_path} -no-pie" ./autogen.sh ${configure_args} --enable-progpath=/usr/bin )
+	( cd ${_driver_dir}/cpca && ./autogen.sh ${configure_args} --enable-progpath=/usr/bin )
+}
+
+do_build() {
+	( cd ${_common_dir} && make )
+	( cd ${_driver_dir} && make )
+}
+
+pre_install() {
+	pushd "lib/libs${_bits}"
+
+	rm "libcanon_slimsfp.so" "libcanon_slimsfp.so.1"
+	rm "libinfor.so" "libinfor.so.1"
+
+	for _so_file in \
+		libcanon_slimsfp.so.1.0.0 \
+		libccpd_utilr.so.1.0.0 \
+		libcomm_stdoutr.so.1.0.0 \
+		libcomm_usbmlportr.so.1.0.0 \
+		libcomm_usbsockr.so.1.0.0 \
+		libcomm_websrvr.so.1.0.0 \
+		libcomm_ncapcaior.so.1.0.0 \
+		libcaio_usb_cdcr.so.1.0.0 \
+		libCUPS_Communicatorr.so.1.0.0 \
+		libCommIsolationr.so.1.0.0 \
+		libcanon_commonr.so.1.0.0 \
+		libinfor.so.1.0.0 \
+		libinfo_analyzer.so.1.0.0 \
+		libcanonncapr.so.1.0.0 \
+		libcnncapcmr.so.1.0.0 \
+		libncapfilterr.so.1.0.0
+	do
+		ln -s ${_so_file} ${_so_file%.1.0.0}
+		ln -s ${_so_file} ${_so_file%.0.0}
+	done
+
+	ln -s libcaepcmsfp.so.1.0 libcaepcmsfp.so.1
+	ln -s libColorGearCsfp.so.2.0.0 libColorGearCsfp.so.2
+	ln -s libColorGearCsfp.so.2.0.0 libColorGearCsfp.so
+
+	popd
+
+	vsed -e 's/cnsetuputil2l\/images\/128x128/icons\/hicolor\/128x128\/apps/' -i cnrdrvcups-utility-5.00/data/cnsetuputil2l.desktop
+	vsed -e 's/Application;Utility/Utility/' -i cnrdrvcups-utility-5.00/data/cnsetuputil2l.desktop
+}
+
+do_install() {
+	for _subdir in "buftool" "cngplp" "cnjbig" "rasterfilter"
+	do
+		( cd ${_common_dir}/${_subdir} && make install DESTDIR=${DESTDIR} COMMON_SUFFIX=2 )
+	done
+
+	for _subdir in "cngplp/files" "cngplp" "cpca" "ppd" "StatusMonitor"
+	do
+		( cd ${_driver_dir}/${_subdir} && make install DESTDIR=${DESTDIR} )
+	done
+
+	pushd "lib/libs${_bits}"
+
+	vbin cnpkmodulencapr
+	vbin cnrsdrvsfp
+	vbin cnsetuputil2l
```

Comment:
These are still binary, what is unfortunate, because driver seems to cross-compile otherwise. Any chance for building from source?

Reply:
Afraid not, although building C applications is beyond my level of expertise and I'm not entirely sure where these come from, because they are not present after extracting the source and before building.